src/main/java/org/skriptlang/reflect/syntax/event/elements/CustomEvent.java

```diff
@@ -75,10 +76,16 @@ public boolean init(Literal<?>[] args, int matchedPattern, SkriptParser.ParseRes
 
     return event.isMarkedContinue();
   }
-
+  
+  @Override
+  public boolean canExecuteAsynchronously() {
+    return isAsync;
+  }
+  
   @Override
   public boolean check(Event e) {
     BukkitCustomEvent bukkitCustomEvent = (BukkitCustomEvent) e;
+    isAsync = bukkitCustomEvent.isAsynchronous() == bukkitCustomEvent.isAsynchroniousAllowed();
```

Comment:
This should not be recomputed at each check, it should be determined at parse time.

Reply:
hm,  can we know sync or async at this moment?
In general, this is something like a fool’s trap so that an asynchronous call is not inserted absolutely everywhere, but idk
```
create section and store it in {_sec}:
  call event (async custom event {_evt}) <----
run {_sec} sync
run {_sec} async
```

Reply:
> hm, can we know sync or async at this moment? In general, this is something like a fool’s trap so that an asynchronous call is not inserted absolutely everywhere, but idk
> 
> ```
> create section and store it in {_sec}:
>   call event (async custom event {_evt}) <----
> run {_sec} sync
> run {_sec} async
> ```

Because `canExecuteAsynchronously` doesn't FORCE async, it just allows it if it is already being run asynchronously.

Reply:
But it is not always necessary for the event to be called without synchronization, or am I misunderstanding something?

Reply:
> But it is not always necessary for the event to be called without synchronization, or am I misunderstanding something?

No, marking as `canExecuteAsynchronously` just means the check() method and the trigger will run on the same thread as the event was called on. If that's async, it's async. If it's the main thread, it stays on the main thread.

If canExecuteAsync... is false, then if the event was called asynchronously, then it gets scheduled to run on the main thread instead.